src/Symfony/Component/DependencyInjection/Loader/XmlFileLoader.php

```diff
@@ -395,7 +397,7 @@ private function processAnonymousServices(\DOMDocument $xml, $file, $defaults)
             foreach ($nodes as $node) {
                 if ($services = $this->getChildren($node, 'service')) {
                     // give it a unique name
-                    $id = sprintf('%d_%s', ++$count, hash('sha256', $file));
+                    $id = sprintf('%d_%s', ++$count, preg_replace('/^.*\\\\/', '', $node->getAttribute('class')).$suffix);
```

Comment:
as a bonus, this adds the final part of the class name to the anonymous service id, to ease reading the id for humans